src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -1492,7 +1492,25 @@ emit_dot (MonoCompile *cfg, MonoClass *klass, MonoType *vector_type, MonoTypeEnu
 static MonoInst*
 emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsig, MonoInst **args)
 {
-	int id = lookup_intrins (sri_vector_methods, sizeof (sri_vector_methods), cmethod);
+	const char *cmethod_name = cmethod->name;
+
+	if (strncmp(cmethod_name, "System.Runtime.Intrinsics.ISimdVector<System.Runtime.Intrinsics.Vector", 70) == 0) {
```

Comment:
It would be nice to create a helper function to share the logic cross `emit_sri_vector` and `emit_sri_vector_t`.

Reply:
I agree, but I had initially viewed that as a larger/riskier refactoring this late in the .NET 9 cycle

Would you be fine with me logging an issue to track the cleanup for .NET 10? -- I think in practice we want `S.Numerics` and `S.R.intrinsics` to share the same importation helper and for any functionality to be differentiated based on the vector size. This is how we do it in RyuJIT since the behavior is typically functionally identical and the only real difference tends to be metadata passed into the IR nodes created (i.e. `Vector4.operator +` and `Vector128.operator +` are identical modulo the type handle; and `Vector64/128/256/512<T>` and `Vector<T>` are all the same modulo the vector size, etc).

Reply:
That sounds great!